libvast/src/system/spawn_counter.cpp

```diff
@@ -32,15 +33,24 @@ spawn_counter(node_actor::stateful_pointer<node_state> self,
               spawn_arguments& args) {
   VAST_TRACE_SCOPE("{}", VAST_ARG(args));
   // Parse given expression.
-  auto expr = parse_expression(args);
-  if (!expr)
-    return expr.error();
+  auto query_result = parse_query(args);
+  if (!query_result)
+    return query_result.error();
+  if (query_result->second) {
+    return caf::make_error(ec::invalid_configuration,
+                           fmt::format("{} parsed a pipeline string which "
+                                       "should "
+                                       "be impossible",
+                                       *self));
+  }
```

Comment:
The parsing isn't impossible, but it does not make sense to execute a pipeline when you're counting results; Instead, that should be expressible as `vast export <format> '... | count'` and return a 1x1 table as a result.

I would suggest the following error message instead:

```
caf::make_error(
  ec::invalid_argument,
  fmt::format(
    "{} failed to spawn counter: the provided query must not contain a pipeline",
    *self
  )
)
```

Reply:
Improved error message in https://github.com/tenzir/vast/pull/2877/commits/d0461071a5c0487d3d189be1978bd17cc56322e9